src/client.py

```diff
@@ -1,32 +1,40 @@
+# coding=utf-8
 """Something client-y."""
 
 import sys
 import socket
 
 
 def client(message):
-
-    message += 'EOF'
+    """Client side of an echo server."""
+    if sys.version_info[0] == 3:
+        try:
+            message = message.decode('utf8')
+        except AttributeError:
+            pass
+    message = str(len(message)) + '.' + message
```

Comment:
You might not want to use periods to delimit your messages. Especially if your message could potentially have a period.